hathor/p2p/manager.py

```diff
@@ -317,11 +334,29 @@ def update_whitelist(self) -> Deferred[None]:
             settings.WHITELIST_URL.encode(),
             Headers({'User-Agent': ['hathor-core']}),
             None)
+        # Twisted Agent does not have a direct way to configure the HTTP client timeout
+        # only a TCP connection timeout. The callLater below is a manual client timeout
+        # that will cancel the deferred in case it's called
+        timeout_call = self.reactor.callLater(10, d.cancel)
+        d.addBoth(self._update_whitelist_cancel_timeout, timeout_call)
         d.addCallback(readBody)
         d.addErrback(self._update_whitelist_err)
         d.addCallback(self._update_whitelist_cb)
         return d
 
+    def _update_whitelist_cancel_timeout(self, param: Union[Failure, Optional[bytes]],
+                                         timeout_call: 'IDelayedCall') -> Union[Failure, Optional[bytes]]:
+        """ This method is always called for both cb and errback in the update whitelist get request deferred.
+            Because of that, the first parameter type will depend, will be a failure in case of errback
+            or optional bytes in case of cb (see _update_whitelist_cb).
+
+            We just need to cancel the timeout call later and return the first parameter,
+            to continue the cb/errback sequence.
+        """
+        if timeout_call.active():
+            timeout_call.cancel()
+        return param
```

Comment:
Why do we have to return it?

Reply:
When you have two callbacks in sequence, the return of the first is the param of the second, doesn't it?

Reply:
It is. I wasn't considering that we would have another callback. But it is always good to return it anyway. Thanks!